stp_concurrent/src/stp_concurrent.cpp

```diff
@@ -0,0 +1,96 @@
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#include "concurrentqueue.hpp"
+using queue = moodycamel::ConcurrentQueue<int>;
+
+class Switch {
+ public:
+  Switch(uint32_t bridgeId) : root_id(bridgeId), bridge_id(bridgeId) {
+    recvQ = std::make_shared<queue>();
+  }
+  friend void link(Switch& a, Switch& b) {
+    a.sendQ.push_back(b.recvQ);
+    b.sendQ.push_back(a.recvQ);
+  }
+  void start() {
+    while (true) {
+      for (auto a : sendQ) {
+        a->enqueue(root_id);
+      }
+
+      // int neighbor;
+      // while (recvQ->try_dequeue(neighbor)) {
+      //   if (neighbor < root_id) root_id = neighbor;
+      //   std::stringstream ss;
+      //   ss << "Switch#" << bridge_id << " received rootID " << neighbor
+      //      << std::endl;
+      //   std::cout << ss.str();
+      // }
+
+      // int neighbors[20];
+      // for (std::size_t count = recvQ->try_dequeue_bulk(neighbors, 20);
+      //      count != 0; --count) {
+      //   if (neighbors[count - 1] < root_id) root_id = neighbors[count - 1];
+      //   std::stringstream ss;
+      //   ss << "Switch#" << bridge_id << " received rootID "
+      //      << neighbors[count - 1] << std::endl;
+      //   std::cout << ss.str();
+      // }
+      uint32_t neighbors[20];
+      std::size_t count;
+      while ((count = recvQ->try_dequeue_bulk(neighbors, 20)) != 0) {
+        for (std::size_t i = 0; i != count; ++i) {
+          if (neighbors[i] < root_id) root_id = neighbors[i];
+          std::stringstream ss;
+          ss << "Switch#" << bridge_id << " received rootID " << neighbors[i]
+             << " :: value of rootID = " << root_id << std::endl;
+          std::cout << ss.str();
+        }
+      }
+      std::this_thread::sleep_for(std::chrono::seconds(10));
+    }
+  }
+
+ private:
+  uint32_t root_id;
+  uint32_t bridge_id;
+  std::shared_ptr<queue> recvQ;
+  std::vector<std::shared_ptr<queue>> sendQ;
```

Comment:
This approach is needed only if no all of the switches get destructed, and remaining ones continue operation.
We don't have to support this, so please use stack allocated object and list/vector of reference wrappers.
If this approach is needed for some other reason, please elaborate.